sos/plugins/etcd.py

```diff
@@ -16,40 +16,38 @@
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
 from sos.plugins import Plugin, RedHatPlugin
-import socket
 
 
 class etcd(Plugin, RedHatPlugin):
 
     """etcd plugin
     """
 
-    def port_report(self):
-        """ Until etcd v2, the daemon listens on port 4001 for backward
-        compatibility of v2.0 eariler. But etcd v3 no loger litens on 
-        port 4001 and it listens on port 2379 only. 
-        To collect information in any version, this plugin should check 
-        which port is available.
-        """
-        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        try:
-            self.sock.connect(("localhost", 2379))
-        except socket.error, msg:
-            self.sock.close()
-            return "4001"
-        self.sock.close()
-        return "2379"
+    def report_url_and_port(self):
+        access_url_port = 'http://localhost:4001'
+        peer_flag = False
+        f = open('/etc/etcd/etcd.conf', 'r')
+
+        for line in f:
+            if line.startswith('ETCD_LISTEN_CLIENT_URLS='):
+                access_url_port = line[len('ETCD_LISTEN_CLIENT_URLS='):-1]
+            elif line.startswith('[peer]'):
+                peer_flag = True
+            elif line.startswith('addr =') and (peer_flag is False):
+                access_url_port = line[len('addr ='):-1]
+
+        f.close()
+        return access_url_port
```

Comment:
This will assume the port is `4001` in the case that configuration is missing or invalid, is that really the best solution? (e.g. why not use 4001 if the version is <= 2, and 2379 if it is 2 or greater).

Also, please fold any "fixup" commits on PR branches back into the main commit (use `git commit --amend` or `git rebase -i`, and then update the branch with `git push --force`).